pkg/kn/commands/root.go

```diff
@@ -78,11 +82,16 @@ Eventing: Manage event subscriptions and channels. Connect up event sources.`,
 func InitializeConfig() {
 	cobra.OnInitialize(initConfig)
 	cobra.OnInitialize(initKubeConfig)
-
+	cobra.OnInitialize(setNamespace)
 }
 
 func initKubeConfig() {
```

Comment:
It would be nice if there were a generic "initKubeConfig" method from the Kubernetes/kubectl libraries.

Add a TODO?

Reply:
I think `clientcmd` will do a lot of this resolution if you don't feed it a kubeCfg file. Not sure if empty string argument will trigger its default resolution, or if you need to use a different method.